src/main/java/duke/component/TaskList.java

```diff
@@ -0,0 +1,51 @@
+package duke.component;
+
+import java.util.ArrayList;
+import java.util.function.Predicate;
+
+import duke.task.Task;
+
+/**
+ * Represents lists of tasks.
+ */
+public class TaskList extends ArrayList<Task> {
+    /**
+     * Creates a list of tasks.
+     */
+    public TaskList() {
+        super();
+    }
+
+    /**
+     * Prints a filtered list based on this list with the given predicate on CLI App.
+     * @param predicate the condition for a task to be printed
+     * @return the number of tasks printed
+     */
+    public int print(Predicate<Task> predicate) {
```

Comment:
Perhaps you could name the method clearer, like printList, to indicate that it is printing a list.